src/components/AppProvider/utils/index.tsx

```diff
@@ -139,6 +138,7 @@ export function withSticky() {
       | React.ComponentClass<OwnProps & WithAppProviderProps> & C
       | React.SFC<OwnProps & WithAppProviderProps> & C,
   ): any & C {
+    // eslint-disable-next-line shopify/react-initialize-state
```

Comment:
I'm not really sure what the correct fix is here so I hand waved it away as this is deep dark typescript magic that I didn't want to mess with

Reply:
I think there is an issue with the order of the generics here. `<{}, OwnProps & WithAppProviderProps>` would mean `{}` is the props and `OwnProps & WithAppProviderProps` is the state which is probably why you're getting this error. Try changing it to `<OwnProps & WithAppProviderProps, never>`.

Reply:
I'm scared, and @AndrewMusgrave was poking around this area in https://github.com/Shopify/polaris-react-deprecated/pull/2274.

I think that PR might solve this, and then because we've got the eslint-comments plugin it'll remind us to delete the ignore line if it is not needed